### DISCUSSION TITLE
Slow startup (~350 ms for `lsp-zero`)

### DISCUSSION THREAD

**Discussion by raivivek** [VonHeikemen/lsp-zero.nvim#23] [Q&A]
Thanks for this project -- it works very well! After messing with various configurations myself, I was able to get started quickly using this plugin. However, there's just one issue. There is a significant startup time when using this plugin ~ 350ms overhead that I profiled using `PackerCompile profile=true` option. Is there a way to reduce that or something is wrong with my configuration? I can post my configuration as well if this timing seems like an outlier based on your experience.

**Comment by VonHeikemen** [comment]
Startup times depends a lot on what servers you have installed. Each language server scans the root directory looking for configuration files. `rust_analyzer` in particular has a very complex process. Also, are you using `ensure_installed()`? Checking those servers are installed takes time (I don't know how much).

The autocompletion plugin can add some miliseconds. Last time I checked, in my configuration it added around 25ms. It depends on your hardware I guess, mine is a very old PC.

Are you using `impatient.nvim`? That plugin can make the `require` function work faster.

**Reply by raivivek** [reply]
Thanks for your reply. I am not using any additional configuration on the top of the minimal config presented in your README. I am mostly looking to work with Python, R, and Bash files, so perhaps there might be a way to improve the load for me by not looking for anything else? And yes, I am also using `impatient.nvim`. For additional information, I am using this configuration on a server that has lots of CPUs and RAM, so I am hoping that resources should not be a challenge, but I will check that to be sure as well.
